client/gutenberg/extensions/slideshow/component.js

```diff
@@ -0,0 +1,154 @@
+						if ( ! image ) {
+							return null;
+						}
+						const { src, alt } = image.props;
+						const figcaption = this.getChildByDataAttribute( child, 'data-is-caption' );
+						const caption = figcaption ? figcaption.props.children : null;
+						const style = {
+							backgroundImage: `url(${ src })`,
+							height: imageHeight,
+						};
+						return (
+							<div className="swiper-slide">
+								<div className="slide-background atavist-cover-background-color" />
+								<div className="wp-block-slideshow-image-container" style={ style } title={ alt } />
+								{ caption && <p className="slideshow-slide-caption">{ caption }</p> }
+							</div>
+						);
+					} ) }
+				</div>
+				<div className="swiper-pagination swiper-pagination-white" ref={ this.paginationRef } />
+				<div className="swiper-button-prev swiper-button-white" ref={ this.btnPrevRef } />
+				<div className="swiper-button-next swiper-button-white" ref={ this.btnNextRef } />
+			</div>
+		);
+	}
+	componentDidMount() {
+		this.buildImageMetadata( this.buildSwiper );
+	}
+	componentDidUpdate( prevProps ) {
+		const { swiperInstance } = this.state;
+		const { align, effect, children } = this.props;
+		if ( children !== prevProps.children ) {
+			this.buildImageMetadata( this.sizeSlideshow );
+		}
+		/* A change in alignment or images only needs an update */
+		if ( align !== prevProps.align || children !== prevProps.children ) {
+			swiperInstance.update();
+		}
+		/* A change in effect requires a full rebuild */
+		if ( effect !== prevProps.effect ) {
+			const activeIndex = swiperInstance.activeIndex;
+			swiperInstance.destroy( true, true );
+			this.buildSwiper( activeIndex );
+		}
+	}
+	buildImageMetadata = callback => {
+		const { children } = this.props;
+		this.setState(
+			{
+				images: Children.map( children, child => {
+					const image = this.getChildByDataAttribute( child, 'data-is-image' );
+					const meta = {
+						width: image.props[ 'data-width' ],
+						height: image.props[ 'data-height' ],
+					};
+					meta.ratio = meta.width / meta.height;
+					return meta;
+				} ),
+			},
+			callback
+		);
+	};
+	buildSwiper = ( initialSlide = 0 ) => {
+		const { effect } = this.props;
+		const swiperInstance = new Swiper( this.slideshowRef.current, {
+			effect: effect,
+			grabCursor: true,
+			init: false,
+			initialSlide: initialSlide,
+			navigation: {
+				nextEl: this.btnNextRef.current,
+				prevEl: this.btnPrevRef.current,
+			},
+			pagination: {
+				clickable: true,
+				el: this.paginationRef.current,
+				type: 'bullets',
+			},
+			preventClicksPropagation: false /* Necessary for normal block interactions */,
+			releaseFormElements: false,
+			setWrapperSize: true,
+			touchStartPreventDefault: false,
+		} );
+		this.setState(
+			{
+				swiperInstance,
+			},
+			() => {
+				swiperInstance.init();
+				this.sizeSlideshow();
+			}
+		);
+	};
+	getChildByDataAttribute = ( parent, attributeName ) => {
```

Comment:
Should this method be named differently? Seems like it would work for non-data attributes (but generic prop names) as well, so maybe `getChildByProp` or `getChildByPropName`?